### PR TITLE
Appflowy for NewsUpdate Editor

### LINKED ISSUE #1384: Appflowy for Text editor

### PR DESCRIPTION
**Pull request by gnunicorn**
This adds the AppflowyEditor for the Text-Updates in our editor for nice WYSIWYG editing.

- [x] fixes #1384 
- [x] refactored update tests for easier reusability
- [x] UI test for html slides
- [x] UI test for multiple html slides
- [x] Text Editor is centered again

All Updates UI tests passing:
image

### CODE REVIEW

File: app/lib/features/news/pages/add_news_page.dart [hunk 1]
```diff
@@ -269,32 +301,23 @@ class _AddNewsState extends ConsumerState<AddNewsPage> {
   }
 
   Widget slideTextPostUI(BuildContext context) {
-    textController.text = selectedNewsPost!.text ?? '';
+    final backgroundColor = ref.watch(
+      newsStateProvider.select((i) => i.currentNewsSlide?.backgroundColor),
+    );
     return Container(
-      padding: const EdgeInsets.symmetric(horizontal: 5.0),
+      padding: const EdgeInsets.all(5.0),
```
Comment (gnunicorn) [inline app/lib/features/news/pages/add_news_page.dart:308]: On the frontend this has now `20` ... should this also be `20` then?

File: app/integration_test/support/util.dart [hunk 2]
```diff
@@ -92,3 +98,29 @@ extension ActerUtil on ConvenientTest {
     await homeKey.tap();
   }
 }
+
+Future<XFile> convertAssetImageToXFile(String assetPath) async {
```
Comment (gnunicorn) [inline app/integration_test/support/util.dart:102]: Primarily a move, but also made it multi-platform by switching to use `p.join`.

File: app/lib/features/news/pages/add_news_page.dart [hunk 3]
```diff
@@ -25,23 +28,52 @@ import 'package:flutter_svg/flutter_svg.dart';
...
-  final textController = TextEditingController();
+class AddNewsState extends ConsumerState<AddNewsPage> {
+  EditorState textEditorState = EditorState.blank();
   NewsSlideItem? selectedNewsPost;
 
   //Build UI
   @override
   Widget build(BuildContext context) {
-    selectedNewsPost = ref.watch(newsStateProvider).currentNewsSlide;
+    ref.listen(newsStateProvider.select((i) => i.currentNewsSlide),
```
Comment (gnunicorn) [inline app/lib/features/news/pages/add_news_page.dart:47]: ref.watch only leads to a re-render. But we need to re-establish the actual `EditorState`, so we instead we are listening to any change of the `currentNewsSlide` (but only that) and do the necessary changes

File: app/lib/features/news/pages/add_news_page.dart [hunk 4]
```diff
@@ -270,36 +302,22 @@ class _AddNewsState extends ConsumerState<AddNewsPage> {
...
 
   Widget slideTextPostUI(BuildContext context) {
-    textController.text = selectedNewsPost!.text ?? '';
+    final backgroundColor = ref.watch(
+      newsStateProvider.select((i) => i.currentNewsSlide?.backgroundColor),
+    );
     return Container(
-      padding: const EdgeInsets.symmetric(horizontal: 5.0),
+      padding: const EdgeInsets.all(5.0),
       alignment: Alignment.center,
```
Comment (gnunicorn) [inline app/lib/features/news/pages/add_news_page.dart:310]: Even tough this has a centered alignment, the editor shows up in the top-left :(

File: app/lib/features/news/pages/add_news_page.dart (hunk 1 above)
Comment (kumarpalsinh25) [reply app/lib/features/news/pages/add_news_page.dart:309]: Yes we can make 20.

File: app/lib/features/news/pages/add_news_page.dart (hunk 4 above)
Comment (kumarpalsinh25) [reply app/lib/features/news/pages/add_news_page.dart:310]: This is probably due to `shrinkWrap` param of `AppFlowyEditor` not working and occupying entire height.

**Review by kumarpalsinh25 — APPROVED** [review]
Great work. Look good overall.

Few minor crash logs I notice while typing on the editor. It will be great if that can be fixed before merge.

Reference:
https://github.com/acterglobal/a3/assets/51526480/f3ca1379-d064-4520-a357-b47aa2058d56

File: app/lib/features/news/pages/add_news_page.dart (hunk 4 above)
Comment (gnunicorn) [reply app/lib/features/news/pages/add_news_page.dart:310]: @kumarpalsinh25 might be related to the fact that it isn't the most outer widget but there are these other overlay widgets around we have to figure out...

**Comment by gnunicorn** [comment]
@kumarpalsinh25 that's some interesting logs. I'll check that. Also seeing an overlay issue at the end of your screen recording, will check on that as well..

**Pull request merged**